Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/mapping/SurveyConfigMapper.kt

```diff
@@ -4,18 +4,69 @@ import dagger.Reusable
-        // TODO
-        return SurveyConfigContainer()
+    override fun map(rawConfig: AppConfigAndroid.ApplicationConfigurationAndroid): SurveyConfig = try {
+        rawConfig.toSurveyConfig()
+    } catch (e: Exception) {
+        Timber.w(e, "Invalid survey config. Treat user survey as disabled")
+        SurveyConfigContainer()
+    }.also { Timber.v("SurveyConfig=%s", it) }
+
+    private fun AppConfigAndroid.ApplicationConfigurationAndroid.toSurveyConfig(): SurveyConfig {
+        val surveyConfig: SurveyConfig
+        if (hasEventDrivenUserSurveyParameters() &&
+            eventDrivenUserSurveyParameters.hasCommon() &&
+            eventDrivenUserSurveyParameters.hasPpac()) {
+            val safetyNetRequirements: SafetyNetRequirements
+            eventDrivenUserSurveyParameters.ppac.also {
+                safetyNetRequirements = SafetyNetRequirementsContainer(
+                    requireBasicIntegrity = it.requireBasicIntegrity,
+                    requireCTSProfileMatch = it.requireCTSProfileMatch,
+                    requireEvaluationTypeBasic = it.requireEvaluationTypeBasic,
+                    requireEvaluationTypeHardwareBacked = it.requireEvaluationTypeHardwareBacked
+                )
+            }
+
+            eventDrivenUserSurveyParameters.common.also {
+                surveyConfig = SurveyConfigContainer(
+                    otpQueryParameterName = it.otpQueryParameterName(),
+                    surveyOnHighRiskEnabled = it.surveyOnHighRiskEnabled,
+                    surveyOnHighRiskUrl = it.surveyOnHighRiskUrl(),
+                    safetyNetRequirements = safetyNetRequirements
+                )
+            }
+        } else {
+            throw ApplicationConfigurationInvalidException(message = "Event driven user survey parameters are missing")
+        }
+
+        return surveyConfig
+    }
+
+    private fun PpddEdusParameters.PPDDEventDrivenUserSurveyParametersCommon.otpQueryParameterName(): String =
+        when (otpQueryParameterName.isNotBlank()) {
+            true -> otpQueryParameterName
+            false -> throw ApplicationConfigurationInvalidException(message = "OTP query parameter name is invalid")
+        }
+
+    private fun PpddEdusParameters.PPDDEventDrivenUserSurveyParametersCommon.surveyOnHighRiskUrl(): String = try {
+        surveyOnHighRiskUrl.toHttpUrl() // Just to validate the given url
+        surveyOnHighRiskUrl
+    } catch (e: Exception) {
+        throw ApplicationConfigurationInvalidException(cause = e, message = "Survey on high risk url is invalid")
     }
 
     data class SurveyConfigContainer(
-        override val safetyNetRequirements: SafetyNetRequirements = SafetyNetRequirementsContainer(),
-        override val enableSurveyLinkOnHighRiskCard: Boolean = true
+        override val otpQueryParameterName: String = "",
+        override val surveyOnHighRiskEnabled: Boolean = false,
+        override val surveyOnHighRiskUrl: String = "",
```

Comment:
Why not use `HttpUrl` here instead of `String`? We are already doing a conversion in the URL validation above...

Reply:
interfaces should be kept as generic as possible and you have to rebuild the url anyay to append the OTP as param.
But anyway, you shall get what you requested 😄 
adjusted.